src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -1394,7 +1398,7 @@ public static BuildResult BuildProjectFileUsingBuildManager(string projectFile,
                     projectFile,
                     new Dictionary<string, string>(),
                     MSBuildConstants.CurrentToolsVersion,
-                    Array.Empty<string>(),
+                    defaultTargets.ToArray() ?? Array.Empty<string>(),
```

Comment:
Needed to include the default targets here since `BuildResult.KeepSpecificTargetResults` relies on the targets passed into `BuildRequestData`.